### ISSUE TITLE
Failed to find nvcc.

### ISSUE THREAD

**Issue by DJDavies2** [ecmwf/atlas#189]
### What happened?

Trying to build atlas is giving this failure in the cmake step:

CMake Error at /spice/data/users/darth/opt/mobbs-52/util/spice_gnu/share/cmake-3.26/Modules/CMakeDetermineCUDACompiler.cmake:180 (message):
  Failed to find nvcc.

  Compiler requires the CUDA toolkit.  Please set the CUDAToolkit_ROOT
  variable.
Call Stack (most recent call first):
  atlas/cmake/features/CUDA.cmake:9 (enable_language)
  atlas/CMakeLists.txt:70 (include)

### What are the steps to reproduce the bug?

As far as I can tell this occurs on any config.  These cmake commands worked a few days ago.

### Version

develop

### Platform (OS and architecture)

Linux

### Organisation

Met Office

**Comment by wdeconinck** [comment]
Hi @DJDavies2 this is a new feature but by default it is OFF, and therefore should not be enabled.
We also encountered this same problem ourselves, when atlas is part of a bundle together with eckit.
eckit (current release 1.26.2) comes in a bundle before atlas, and has the CUDA feature with "DEFAULT ON". That then initialises "ENABLE_CUDA ON" and makes atlas configuration fail.
The workaround is either to explicitly set "-DENABLE_CUDA=OFF" on the configuration line, or update eckit its develop branch where I made a commit last week to turn the DEFAULT OFF: https://github.com/ecmwf/eckit/commit/f45d23935755b12aeff42f2183a214835a80cb31 .
